architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,247 @@
+# Radius UCP Component Threat Model
+
+- **Author**: ytimocin
+
+## Overview
+
+This document provides a threat model for the Radius UCP (Universal Control Plane) component. It identifies potential security threats to this critical part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+One of the most important goals of Radius is to be cloud agnostic and to enable portability of applications across platforms and environments. Therefore, the Radius UCP (Universal Control Plane) component helps the user apply ARM capabilities to non-Azure hosting models like AWS, GCP, and Kubernetes.
+
+In brief, UCP is a lightweight proxy that stands in front of the resource providers like Applications RP, AWS, and Azure. Based on the requests UCP gets, it forwards those requests to the appropriate service.
+
+## Terms and Definitions
+
+| Term                    | Definition                                                                                                                                  |
+| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+| Radius Tracked Resource | Ex: `/planes/radius/local/resourceGroups/test-group/providers/System.Resources/resources/test-app-303153687ee5adbcf353bc6c2caa4373f31e04c6` |
+
+## System Description
+
+An example flow:
+
+1. Radius user runs `rad deploy app.bicep`. `app.bicep` includes a Radius environment, a Radius application, a Redis Cache on AWS.
+2. ARM JSON Template is generated from the `app.bicep` file.
+3. CLI makes a PUT call to UCP: `/planes/radius/local/resourcegroups/{rg}/providers/Microsoft.Resources/deployments/{name}`
+4. UCP forwards that request to the Deployment Engine: `/planes/radius/local/resourcegroups/{rg}/providers/Microsoft.Resources/deployments/{name}`
+5. Deployment Engine makes 3 calls back to the UCP:
+   1. PUT for the Radius environment.
+   2. PUT for the Radius application.
+   3. PUT for the Redis Cache on AWS.
+6. UCP proxies these requests to the necessary resource providers and then starts polling the statuses of these operations.
+7. Deployment Engine also polls the operation statuses and results for these 3 resources.
```

Comment:
I think this is a good data flow that captures the majority of UCP's functionality and interactions with other systems.